### DISCUSSION TITLE
First request on Vercel with dynamic routes

### DISCUSSION THREAD

**Discussion by dekisr** [vercel/next.js#24157] [Help]
Hello there,

I'm trying some dynamic routes on Vercel, they are not working as I would like.

Assuming this code:
```javascript
export async function getStaticPaths() {
  return { paths: [], fallback: true };
}

export async function getStaticProps() {
  const data = await someFetch();
  if (!data) return { notFound: true};

  return {
    props: { data },
    revalidate: 30,
  };
}
```

Every **first** request to any dynamic path are going to 404.
And then all subsequent requests are working fine.

Testing on Vercel deploy preview

**Comment by imadatyatalah** [comment]
Take a look at [Fallback pages](https://nextjs.org/docs/basic-features/data-fetching#fallback-pages)
I Hope it will help you.

**Comment by dekisr** [comment]
Hey, thanks @imadatyatalah 

I'll take a look! But I figured out, maybe it's something else.

On our `next.config.js` we have a rewrite :
```javascript
{
  source: '/:path*',
  destination: '/:path*',
},
```

We are using this because we were following the [Incremental adoption of Next.js](https://nextjs.org/docs/api-reference/next.config.js/rewrites#incremental-adoption-of-nextjs) from docs.

But it's changing our `context.params` behavior,
Suppose the `[testSlug].js` dynamic page:
```javascript
import { useRouter } from 'next/router'

export async function getStaticPaths() {
  return { paths: [], fallback: true }
}

export async function getStaticProps({ params }) {
  const { testSlug } = params
  return testSlug ? { props: { testSlug } } : { notFound: true }
}

export default function Test({ testSlug }) {
  const { isFallback } = useRouter()
  return isFallback ? <h1>fallback...</h1> : <h1>{testSlug}</h1>
}

```

On the first request, we are receiving the `testSlug` prop as a string `"[testSlug].json"`. And then all subsequent requests with previously used slugs are working fine.

Here some example to check : https://supportrequest-rewrites.vercel.app/test/trysomenewslug

**Reply by imadatyatalah** [reply]
Honestly i don't know where the problem is, I'll clone that [repo](https://github.com/dekisr/supportrequest-rewrites) and try to solve your problem.

**Reply by imadatyatalah** [reply]
it runs locally without any problems

**Reply by dekisr** [reply]
Yes, running locally we have no issues. It's occurring after we deploy to Vercel. And thank you @imadatyatalah! We are trying some workaround logic on `getStaticProps`, but maybe it's not the most correct way to solve this.

**Reply by imadatyatalah** [reply]
the problem is from `next.config.js` file, I've remove it and it's working now you can [test it](https://supportrequest-rewrites-theta.vercel.app/test/anotherRoute)

**Reply by dekisr** [reply]
Yes, I already mentioned above, we are using the rewrites config, following the [Incremental adoption of Next.js](https://nextjs.org/docs/api-reference/next.config.js/rewrites#incremental-adoption-of-nextjs) from docs
```
{
  source: '/:path*',
  destination: '/:path*',
},
```
That part is causing the strange behavior on `context.params` of our `getStaticProps` dynamic pages
